webui/src/components/form/InputNumber.tsx

```diff
@@ -0,0 +1,105 @@
+import type { ChangeEvent } from 'react'
+import { useCallback, useMemo } from 'react'
+import type { InputProps } from 'baseui/input'
+import { useStyletron } from 'baseui'
+import { Input as BaseUIInput } from 'baseui/input'
+import { Slider as BaseUISlider } from 'baseui/slider'
+
+export interface IInputNumberProps extends Omit<InputProps, 'max' | 'min' | 'step' | 'value' | 'onChange'> {
+  value: number
+  step?: number
+  minimum?: number
+  maximum?: number
+  exclusiveMinimum?: number
+  exclusiveMaximum?: number
+  onChange?: (value: number) => void
+}
+
+export default function InputNumber({
+  value,
+  step = 1,
+  onChange,
+  minimum,
+  maximum,
+  exclusiveMinimum,
+  exclusiveMaximum,
+  ...restProps
+}: IInputNumberProps) {
+  const [css] = useStyletron()
+  const min = useMemo(() => {
+    if (exclusiveMinimum !== undefined)
+      return exclusiveMinimum
+    if (minimum !== undefined)
+      return minimum + step
+    return undefined
+  }, [minimum, exclusiveMinimum])
+  const max = useMemo(() => {
+    if (exclusiveMaximum !== undefined)
+      return exclusiveMaximum
+    if (maximum !== undefined)
+      return maximum - step
+    return undefined
+  }, [maximum, exclusiveMaximum])
+  // component show slider when both max and min exist
+  const showSlider = useMemo(() => min !== undefined && max !== undefined, [min, max])
+  const handleInputChange = useCallback((e: ChangeEvent<HTMLInputElement | HTMLTextAreaElement>) => {
+    // e.target.valueAsNumber is NaN when input's value is '-' or empty string
+    // TODO: typescript unsupport type of ChangeEvent valueAsNumber, but most browsers support it
+    // https://caniuse.com/mdn-api_htmlinputelement_valueasnumber
+    let value: number = (e.target as any).valueAsNumber
+    if (Number.isNaN(value))
+      return onChange?.(e.target.value as unknown as number)
+
+    // check range
+    if (max && value > max)
+      value = max
+    if (min && value < min)
+      value = min
```

Comment:
Need to check against the original `maximum/minimum/exclusiveMaximum/exclusiveMinimum`